napari/util/io.py

```diff
@@ -1,66 +1,27 @@
 import os
+from glob import glob
 
 import numpy as np
 from skimage import io
+from skimage.io.collection import alphanumeric_key
 
-dask_available = True
-try:
-    from dask import array as da
-except ImportError:
-    dask_available = False
-
-
-def imread(filenames, *, use_dask=None, stack=True):
-    """Read image files and return an array.
-
-    If multiple images are selected, they are stacked along the 0th axis.
-
-    Parameters
-    -------
-    filenames : list
-        List of filenames to be opened
-    use_dask : bool
-        Whether to use dask to create a lazy array, rather than NumPy.
-        Default is None, which is interpreted as "use if available". If set
-        to True and dask is not installed, this function
-    stack : bool
-        Whether to stack the images in multiple files into a single array. If
-        False, a list of arrays will be returned.
-
-    Returns
-    -------
-    image : array
-        Array of images
-    """
-    if dask_available and use_dask is None:
-        use_dask = True
-    if not dask_available and use_dask:
-        raise ValueError('Dask array requested but dask is not installed.')
-    images = [io.imread(filename) for filename in filenames]
-    if len(images) == 1:
-        image = images[0]
-    else:
-        if use_dask:
-            image = da.stack(images)
-        else:
-            image = np.stack(images)
-
-    return image
+from dask import delayed
+from dask import array as da
 
 
 def magic_read(filenames, *, use_dask=None, stack=True):
     """Dispatch the appropriate reader given some files.
 
-    The files are assumed to all have the same type.
+    The files are assumed to all have the same shape.
```

Comment:
Unfortunately when I do this I'm getting an error message due to our contrast limits calculation :-(
```python
Traceback (most recent call last):
  File "/Users/nicholassofroniew/Github/napari/napari/_qt/qt_viewer.py", line 222, in _open_images
    self._add_files(filenames)
  File "/Users/nicholassofroniew/Github/napari/napari/_qt/qt_viewer.py", line 240, in _add_files
    self.viewer.add_image(image, rgb=is_rgb(image.shape))
  File "/Users/nicholassofroniew/Github/napari/napari/components/viewer_model.py", line 470, in add_image
    visible=visible,
  File "/Users/nicholassofroniew/Github/napari/napari/layers/image/image.py", line 191, in __init__
    self._contrast_limits_range = calc_data_range(input_data)
  File "/Users/nicholassofroniew/Github/napari/napari/util/misc.py", line 282, in calc_data_range
    min_val = reduced_data.min()
  File "/anaconda3/lib/python3.7/site-packages/numpy/core/_methods.py", line 34, in _amin
    return umr_minimum(a, axis, None, out, keepdims, initial, where)
ValueError: operands could not be broadcast together with shapes (800,3) (812,3)
```

If I comment out the contrast_limit stuff it works beautifully. I know this is a problem we've been facing for a while - the offending line is this horrific one that assumes all the planes are the same shape when concatenating them
https://github.com/napari/napari/blob/e61943347401ff6fe9da669513ae86ea7ac99a19/napari/util/misc.py#L279
which is totally unnecessary as you could just take the min / max on each separately and then take the min again, the following change from line 275 on solves it
```python
        top_plane = np.asarray(data[tuple(top_plane_idx)])
        bottom_plane = np.asarray(data[tuple(bottom_plane_idx)])
        middle_plane = np.asarray(data[tuple(middle_plane_idx)])
        min_val = np.array([top_plane.min(), bottom_plane.min(), middle_plane.min()]).min()
        max_val = np.array([top_plane.max(), bottom_plane.max(), middle_plane.max()]).max()
    else:
        min_val = data.min()
        max_val = data.max()
```